### PR TITLE
Feature/favoriting functionality cr

### LINKED ISSUE #26: Add functionality for favoriting

### PR DESCRIPTION
**Pull request by Fordo29**
## What Features are Being Added?
Features add are the Post/ Delete to favorites database and Patch to books database

## What is Being Refactored?
Post object was trimmed to post correctly

## Where should the reviewer start?
BookDetails.js, apiCalls.js, utils.js

## How should this be manually tested?
Go to the single book details page and click to add to favorites - the button should change

## What are the relevant tickets?
NA

## Any Other Notes/Questions/Problems?
Not anymore

### CODE REVIEW

File: src/apiCalls.js [hunk 1]
```diff
@@ -20,8 +20,17 @@ export const apiCalls = {
...
     }).then(response => response.json())
   },
-  deleteFromFavorites(id) {
-    return fetch(`https://book-club-api-2110.herokuapp.com/api/v1/favorites/${id}`, { method: 'DELETE' })
+  deleteFromFavorites(isbn) {
+    return fetch(`https://book-club-api-2110.herokuapp.com/api/v1/favorites/${isbn}`, { method: 'DELETE' })
       .then(response => response.json())
-  } 
+  }, 
+  updateFavStatus(selectedBook) {
```
Comment (krdurrett) [inline src/apiCalls.js:27]: great logic for this patch

Comment (cagallo) [inline src/apiCalls.js:27]: Great PATCH request, looks clean!

File: src/apiCalls.js [hunk 2]
```diff
@@ -20,8 +20,17 @@ export const apiCalls = {
...
+    return fetch(`https://book-club-api-2110.herokuapp.com/api/v1/books/${selectedBook.isbn}`, {
+      method: 'PATCH',
+      body: JSON.stringify({isFavorited: !(selectedBook.isFavorited === 'true')}),
```
Comment (ectweitmann) [inline src/apiCalls.js:30]: Nice work around!

File: src/components/BookDetails.js [hunk 3]
```diff
@@ -1,27 +1,64 @@
...
     apiCalls.getSingleBook(this.props.isbn)
       .then(bookData => {
         const cleanedBookData = cleanBookData(bookData[0]);
-        this.setState({ selectedBook: cleanedBookData });
+        this.setState({ selectedBook: cleanedBookData});
       })
       .catch(error => this.setState({ error: error.message }))
   }
 
+  handleDelete = () => {
```
Comment (krdurrett) [inline src/components/BookDetails.js:25]: these helper functions for adding and deleting favs look great

File: src/components/BookDetails.js [hunk 4]
```diff
@@ -1,27 +1,64 @@
...
+        .then(data => {
+          const cleanedBookData = cleanBookData(data[0])
+          console.log('add to fav', cleanedBookData)
+          this.setState({ selectedBook: cleanedBookData })
+        })
+    })
+    .catch(error => this.setState({ error: error.message }))
+  }
+
+  determineButton = () => {
```
Comment (krdurrett) [inline src/components/BookDetails.js:51]: This is a great helper function to determine what the button is supposed to do/show

File: src/components/BookDetails.js (hunk 3 above)
Comment (cagallo) [inline src/components/BookDetails.js:25]: This looks great!! Good to do promise.all for these!

File: src/utils.js [hunk 5]
```diff
@@ -11,6 +11,17 @@ export const cleanBookData = (book) => {
     book_image: book['book_image'],
     author: book.author,
     isFavorited: book.isFavorited
-  } 
+  }   
+}
+
+export const trimBookData = (book) => {
```
Comment (krdurrett) [inline src/utils.js:17]: great helper function to clean the data for the post

**Review by cagallo — APPROVED** [review]

**Review by krdurrett — APPROVED** [review]
This button functionality looks great!

**Review by ectweitmann — APPROVED** [review]

**Pull request merged**